tensorboard/webapp/widgets/histogram/histogram_v2_component.ts

```diff
@@ -239,8 +231,25 @@ export class HistogramV2Component
 
   onResize() {
     this.updateClientRects();
-    this.updateChart();
-    this.changeDetector.detectChanges();
+    this.updateChartIfVisible();
+  }
+
+  onVisibilityChange({visible}: {visible: boolean}) {
+    this.domVisible = visible;
+    if (!visible) return;
+
+    if (!this.domInitialized) {
+      this.domInitialized = true;
+      this.changeDetector.detectChanges();
+      fromEvent<MouseEvent>(this.main.nativeElement, 'mousemove', {
```

Comment:
By moving the `fromEvent` into `onVisibilityChange`, would this be triggering a new `addEventListener` every time a histogram is scrolled into view?  I think the `ngAfterViewInit` might be a preferred place for this, if we don't want unbounded # of listeners.

Reply:
To prevent constant subscription, I am doing this once using `domInitialized`.

Reply:
d'oh that's pretty clear upon re-read, sorry